apps/designer/app/designer/shared/assets/types.ts

```diff
@@ -17,20 +17,28 @@ export type UploadingClientAsset = {
   preview: PreviewAsset;
 };
 
+/**
+ * Used for optimistic UI only
+ **/
 export type DeletingClientAsset = {
   status: "deleting";
   asset: Asset;
   preview: PreviewAsset | undefined;
 };
 
 /**
- * Client side we need more information about the asset than the server provides.
+ * On the client side, we need more information about the asset than the server provides.
  */
 export type ClientAsset =
   | UploadedClientAsset
   | UploadingClientAsset
   | DeletingClientAsset;
 
+/**
+ * Assets that can be shown in the UI
+ */
+export type ActiveAsset = UploadedClientAsset | UploadingClientAsset;
```

Comment:
Maybe RenderableAsset would be more clear

Reply:
Yep better